src/api/java/com/minecolonies/api/crafting/ItemStorage.java

```diff
@@ -269,12 +269,18 @@ public boolean equals(final Object o)
 
 
         return stack.isItemEqual(that.getItemStack())
-                 && (this.shouldIgnoreDamageValue || that.getDamageValue() == this.getDamageValue())
+                 && (this.shouldIgnoreDamageValue || that.shouldIgnoreDamageValue || that.getDamageValue() == this.getDamageValue())
                  && (this.shouldIgnoreNBTValue || that.shouldIgnoreNBTValue
                        || (that.getItemStack().getTag() == null && this.getItemStack().getTag() == null)
                        || (that.getItemStack().getTag() != null && that.getItemStack().getTag().equals(this.getItemStack().getTag())));
     }
 
+    public boolean matchDefinitionEquals(ItemStorage that)
```

Comment:
jdoc

Reply:
Again? Oh, right. I did it once, removed it during our discussion yesterday about how to resolve this, and then put it back again, forgetting the Jdoc this time. 

That's my story, and I'm sticking to it.